api/src/main/java/org/openmrs/module/locationbasedaccess/PatientSearchAdvise.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.locationbasedaccess;
+
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInvocation;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.PersonAttribute;
+import org.openmrs.api.context.Context;
+import org.springframework.aop.Advisor;
+import org.springframework.aop.support.StaticMethodMatcherPointcutAdvisor;
+import java.lang.reflect.Method;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * It will be used to implement the AOP Methods for PatientServiceImpl class
+ */
+public class PatientSearchAdvise extends StaticMethodMatcherPointcutAdvisor implements Advisor {
+
+    protected static final Log log = LogFactory.getLog(PatientSearchAdvise.class);
+
+    @Override
+    public boolean matches(Method method, Class targetClass) {
+        if (method.getName().equals("getPatients")) {
+            return true;
+        }
+        else if (method.getName().equals("getPatient")) {
```

Comment:
Does the patient search widget use getPatient() or getPatients()?

Reply:
It used this both methods for the patient search. 
1. While we are searching for the patient search widget, it will use this getPatients()
2. While we get the last access patients (initial view without searching), it will use getPatient() method.

Reply:
No @dkayiwa, Both are executed without any interactions.

Reply:
So where does getPatient get the id of the patient to fetch?

Reply:
When we enter to the patient search page, It will automatically display some rows of the patients which are from mostRecentRegistrationEncounters. Those mostRecentRegistrationEncounters is fetched using the getPatient(id) method.  So the patientIds fetched from those most recent encounters. 

Then when we are start to search for the patients, the patient search page start to fetch the patient through the REST API, then its depend on the getPatients() method. So I wanted to track those two methods to restrict the patient search based on the locations.

Reply:
Let me put it this way, what happens if we only deal with getPatients()?

Reply:
Then we will get all the patients in the view when come to the findPatient dashboard first time and we haven't any values in the search box. (It will fetch all the recent encounters from all users without considering the access Locations)

Reply:
To confirm this, can you point me to the portion of code which fills the patient list on the dashboard by using the getPatient() method instead of getPatients()?

Reply:
Let's have a look at here @dkayiwa 
* https://github.com/openmrs/openmrs-module-registrationapp/blob/master/omod/src/main/webapp/pages/findPatient.gsp#L66
* https://github.com/openmrs/openmrs-module-registrationapp/blob/master/omod/src/main/java/org/openmrs/module/registrationapp/page/controller/FindPatientPageController.java#L116

Those are used to fetch the mostRecentRegistrationEncounters, and the encounter used getPatient() method to retrieve the patient Info.

Reply:
For this pull request, let us deal with only getPatients()

Reply:
Then, I have removed getPatient() from the implementation.